factory/src/test/kotlin/io/qalipsis/core/factory/campaign/CampaignTimeoutKeeperTest.kt

```diff
@@ -0,0 +1,245 @@
+                campaignKey = "my-campaign",
+                broadcastChannel = "broadcast-channel",
+                feedbackChannel = "feedback-channel",
+                speedFactor = 12.0,
+                startOffsetMs = 1212,
+                scenarios = mapOf(
+                    "scenario-1" to ScenarioConfiguration(
+                        123,
+                        RegularExecutionProfileConfiguration(123, 3534)
+                    ), "scenario-2" to ScenarioConfiguration(5432, DefaultExecutionProfileConfiguration())
+                ),
+                hardTimeout = null,
+                softTimeout = Instant.now().plusSeconds(1),
+                assignments = listOf(
+                    FactoryScenarioAssignment("my-scenario-1", listOf("dag-1", "dag-2")),
+                    FactoryScenarioAssignment("my-scenario-2", listOf("dag-3", "dag-4"))
+                )
+            )
+            val campaignTimeoutKeeper = CampaignTimeoutKeeper(factoryChannel, this)
+            val latch = Latch(true)
+            coEvery { factoryChannel.publishFeedback(any()) } coAnswers { latch.release() }
+
+            // when
+            campaignTimeoutKeeper.init(campaign)
+            latch.await()
+
+            //then
+            coVerifyOrder {
+                factoryChannel.publishFeedback(
+                    CampaignTimeoutFeedback(
+                        campaignKey = "my-campaign",
+                        status = FeedbackStatus.COMPLETED,
+                        hard = false,
+                        errorMessage = "The soft timeout was reached"
+                    )
+                )
+            }
+            confirmVerified(factoryChannel)
+        }
+
+    @Test
+    fun `should not publish any timeout feedback when no timeout is specified`() =
+        testDispatcherProvider.run {
+            val campaign = Campaign(
+                campaignKey = "my-campaign",
+                broadcastChannel = "broadcast-channel",
+                feedbackChannel = "feedback-channel",
+                speedFactor = 12.0,
+                startOffsetMs = 1212,
+                scenarios = mapOf(
+                    "scenario-1" to ScenarioConfiguration(
+                        123,
+                        RegularExecutionProfileConfiguration(123, 3534)
+                    ), "scenario-2" to ScenarioConfiguration(5432, DefaultExecutionProfileConfiguration())
+                ),
+                hardTimeout = null,
+                softTimeout = null,
+                assignments = listOf(
+                    FactoryScenarioAssignment("my-scenario-1", listOf("dag-1", "dag-2")),
+                    FactoryScenarioAssignment("my-scenario-2", listOf("dag-3", "dag-4"))
+                )
+            )
+            val campaignTimeoutKeeper = CampaignTimeoutKeeper(factoryChannel, this)
+
+            // when
+            campaignTimeoutKeeper.init(campaign)
+
+            //then
+            coVerifyNever {
```

Comment:
You put @KTestable on the jobs properties in the class, perhaps you could check that they are still null.